chaincode/javascript/lib/medical.js

```diff
@@ -0,0 +1,226 @@
+
+    //add doctors under consentTo
+
+    /**
+     *
+     * @param {Context} ctx
+     * @param {string} patientRecordId
+     * @param {string} targetName
+     * @param {string} medicalRecordId
+    
+     */
+    async providingConsent(ctx, patientRecordId, targetName, medicalRecordId) {
+        //extracting the caller record
+        const recordAsBytes = await ctx.stub.getState(patientRecordId);
+        if (!recordAsBytes || recordAsBytes.length === 0) {
+            throw new Error(`${patientRecordId} does not exist`);
+        }
+        //converting into string
+        const record = JSON.parse(recordAsBytes.toString());
+
+        //adding target into caller's consentTo
+        const permission = record.medicalRecord[
+            medicalRecordId
+        ].consentTo.filter((doc) => {
+            return doc.id === targetName;
+        });
+        if (!permission || permission.length === 0) {
+            record.medicalRecord[medicalRecordId].consentTo.push(targetName);
+        }
+
+        await ctx.stub.putState(
+            patientRecordId,
+            Buffer.from(JSON.stringify(record))
+        );
+        return true;
+    }
+    /**
+     *
+     * @param {Context} ctx
+     * @param {string} patientRecordId
+     * @param {string} targetName
+     * @param {string} medicalRecordId
+    
+     */
+    async removingConsent(ctx, patientRecordId, targetName, medicalRecordId) {
+        //extracting the caller record
+        const recordAsBytes = await ctx.stub.getState(patientRecordId);
+        if (!recordAsBytes || recordAsBytes.length === 0) {
+            throw new Error(`${patientRecordId} does not exist`);
+        }
+        //converting into string
+        const record = JSON.parse(recordAsBytes.toString());
+
+        //adding target into caller's consentTo
+        const permission = record.medicalRecord[
+            medicalRecordId
+        ].consentTo.filter((doc) => {
+            return doc.id != targetName;
+        });
+        record.medicalRecord[medicalRecordId].consentTo = permission;
+
+        await ctx.stub.putState(
+            patientRecordId,
+            Buffer.from(JSON.stringify(record))
+        );
+        return true;
+    }
+    /**
+     * @param {Context} ctx
+     * @param {string} patientId
+     * @param {string} medicalInfo
+     */
+    async writePatientMedicalInfo(ctx, patientId, medicalRecordObj) {
+        //extract the patient record
+        //will receive the record in form of bytes
+        const {
+            medicalRecordId,
+            uploadedBy,
+            medicalRecordData,
+            consentTo,
+        } = medicalRecordObj;
+        const recordAsBytes = await ctx.stub.getState(patientId);
+        if (!recordAsBytes || recordAsBytes.length === 0) {
+            throw new Error(`${patientId} does not exist`);
+        }
+        //converting into string
+        const record = JSON.parse(recordAsBytes.toString());
+
+        //checking for permission
```

Comment:
I checked the assignment document and there was no restriction on adding data. I guess when you are Health Entity you can add for any user.